### DISCUSSION TITLE
Updating the parent component's state inside Recharts Issues

### DISCUSSION THREAD

**Discussion by victorias** [recharts/recharts#4223] [Q&A]
Hey, wondering if people have seen stuff like this:

```client.js:1 Warning: Cannot update a component (`ParentChart`) while rendering a different component (`AreaChart`). To locate the bad setState() call inside `AreaChart`,```

This often happens when we call setState within ReCharts. Have others seen this and what ideas do you have about it?

**Comment by ckifer** [comment]
> when we call setState within Recharts.

What do you mean by this? Can you make a sandbox that reproduces the issue or give an example?
